src/nfa/nfa.cc

```diff
@@ -1419,31 +1410,35 @@ Nfa::const_iterator& Nfa::const_iterator::operator++()
 
     // out of state set
     ++(this->tlIt);
-    const Moves& tlist = this->nfa->get_moves_from(this->trIt);
+    const Post& tlist = this->nfa->get_moves_from(this->trIt);
     assert(!tlist.empty());
     if (this->tlIt != tlist.end())
     {
-        this->ssIt = this->tlIt->states_to.begin();
+        this->ssIt = this->tlIt->targets.begin();
 
         this->refresh_trans();
         return *this;
     }
 
     // out of transition list
     ++this->trIt;
-    assert(!std::all_of(this->nfa->transition_relation.begin(), this->nfa->transition_relation.end(),
-                        [](const auto& trans) {return trans.size() == 0;}));
-    while (this->trIt < this->nfa->transition_relation.size() &&
-            this->nfa->get_moves_from(this->trIt).empty())
+    bool any_nonempty = false;
+    for (size_t i = 0; i < this->nfa->delta.size() && !any_nonempty; ++i) {
+        any_nonempty = !this->nfa->delta[i].empty();
```

Comment:
Again, I think we could short-circuit here

Reply:
refactored.